pkg/server/internal/resultset/resultset.go

```diff
@@ -142,3 +143,25 @@ func (trs *tidbResultSet) FieldTypes() []*types.FieldType {
 func (trs *tidbResultSet) SetPreparedStmt(stmt *core.PlanCacheStmt) {
 	trs.preparedStmt = stmt
 }
+
+// TryDetach creates a new `ResultSet` which doesn't depend on the current session context.
+func (trs *tidbResultSet) TryDetach() (ResultSet, bool, error) {
+	detachableRecordSet, ok := trs.recordSet.(sqlexec.DetachableRecordSet)
+	if !ok {
+		return nil, false, nil
+	}
+
+	recordSet, detached, err := detachableRecordSet.TryDetach()
+	if !detached {
```

Comment:
Do we need to process errors firstly? Or just:

```
if !detached || err != nil {
    return recordSet, detached, err
}
```

Reply:
Done.